libgnucash/app-utils/gnc-option-impl.cpp

```diff
@@ -224,7 +224,7 @@ GncOptionCommodityValue::reset_default_value()
 bool
 GncOptionCommodityValue::is_changed() const noexcept
 {
-    return m_namespace == m_default_namespace && m_mnemonic == m_default_mnemonic;
+    return m_namespace == m_default_namespace && m_mnemonic != m_default_mnemonic;
```

Comment:
This should be either
```
return !(m_namespace == m_default_namespace && m_mnemonic == m_default_mnemonic);
```
or
```
return m_namespace != m_default_namespace || m_mnemonic != m_default_mnemonic;
```
To cover the possibility that two namespaces have the same mnemonic.